### ISSUE TITLE
Changing a shared formula's first cell incorrectly fills in gaps in the shared formula range

### ISSUE THREAD

**Issue by craigbrown** [EPPlusSoftware/EPPlus#477]
Excel can sometimes store similar formulas as "shared formulas" without telling you.

In the attached workbook, I have a worksheet which has a shared formula in `G1:G6`, but no formula in `G5`.

If I delete the formula from `G1` (but not from any of the other cells), the formula is correctly removed from `G1` - but it now appears in cell `G5` where there should be no formula.

```csharp
[TestMethod]
public void RemoveFormulaFromFirstCellOfSharedFormulaWithGap()
{
    using (var pck = OpenPackage("TestSharedFormula.xlsx"))
    {
        // Get the worksheet
        var wks = pck.Workbook.Worksheets["Sheet1"];
        
        // Check that G1:G6 have formulas, except for G5
        Assert.AreEqual("SUM(A1:F1)", wks.Cells["G1"].Formula);
        Assert.AreEqual("SUM(A2:F2)", wks.Cells["G2"].Formula);
        Assert.AreEqual("SUM(A3:F3)", wks.Cells["G3"].Formula);
        Assert.AreEqual("SUM(A4:F4)", wks.Cells["G4"].Formula);
        Assert.AreEqual("", wks.Cells["G5"].Formula);
        Assert.AreEqual("SUM(A6:F6)", wks.Cells["G6"].Formula);
        
        // Remove the formula from G1
        wks.Cells["G1"].Formula = null;
        
        // Check that the formula was removed, and the other formulas are unchanged
        Assert.AreEqual("", wks.Cells["G1"].Formula);
        Assert.AreEqual("SUM(A2:F2)", wks.Cells["G2"].Formula);
        Assert.AreEqual("SUM(A3:F3)", wks.Cells["G3"].Formula);
        Assert.AreEqual("SUM(A4:F4)", wks.Cells["G4"].Formula);
        Assert.AreEqual("", wks.Cells["G5"].Formula);
        Assert.AreEqual("SUM(A6:F6)", wks.Cells["G6"].Formula);
    }
}
```
[TestSharedFormula.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/7092474/TestSharedFormula.xlsx)

**Comment by JanKallman** [comment]
When adjusting a shared formula address the index is rewritten for the range, not checking if it had the sf-index.
I'll provide a fix.

**Comment by JanKallman** [comment]
Fixed in version 5.7.5
